syntax = "proto2";

import "ignition/msgs/vector2d.proto";
import "ignition/msgs/vector3d.proto";

package smartmouse.msgs;

message RobotDescription {
    required Polygon2D footprint = 1;
    required ignition.msgs.Vector2d left_wheel = 2;
    required ignition.msgs.Vector2d right_wheel = 3;
    optional ignition.msgs.Vector3d cog = 4;
}

message Polygon2D {
    repeated ignition.msgs.Vector2d pts = 1;
}
